queue/byte_queue_test.go

```diff
@@ -90,3 +95,71 @@ func (suite *ByteQueueSuite) TestPullDelegatesToProcessor() {
 
 	suite.Assert().Equal([]byte("bar"), <-q.In())
 }
+
+func (suite *ByteQueueSuite) TestConcatsSuccessfully() {
```

Comment:
I don't think the responsibility of this test is to check that concatenation was successful, rather that the ByteQueue delegates to the processor.